lib/facter/kmod.rb

```diff
@@ -0,0 +1,29 @@
+Facter.add(:kmods) do
+  confine :kernel => 'Linux'
+  confine File.exists?('/sys/module')
+
+  kmod = {}
+
+  setcode do
+    Dir.foreach('/sys/module') do |directory|
+      next if directory == '.' or directory == '..'
+
+      kmod[directory] = {}
+      kmod[directory]['parameters'] = {}
+      kmod[directory]['used_by'] = []
+
+      Dir.foreach("/sys/module/#{directory}/parameters") do |param|
+        next if param == '.' or param == '..'
```

Comment:
Using `Dir.glob("/sys/module/#{directory}/parameters/*")` will save you from having to filter out `.` and `..`